bucket_map/src/bucket.rs

```diff
@@ -268,10 +268,11 @@ impl<T: Clone + Copy> Bucket<T> {
     pub fn try_write(
         &mut self,
         key: &Pubkey,
-        data: &[T],
+        data: impl Iterator<Item = &'b T>,
```

Comment:
nit: maybe we can combine this into one struct instead of two separate parameters? 
```
struct CountIter<'b, T: Clone + Copy + 'b> {
  iter: impl Iterator<Item = &'b T>,
  len: usize,
}
```